### DISCUSSION TITLE
How can I undo an object if it has been erased using the Eraser tool?

### DISCUSSION THREAD

**Discussion by luongtu996** [fabricjs/fabric.js#8749] [Q&A]
example: 
If i have a rectangle and i used an eraser brush to delete a part of it, and you want to undo the original rect

Step 1.
![Screenshot 2023-03-02 at 09 53 10](https://user-images.githubusercontent.com/26574116/222319051-85273f76-3be7-46c5-959b-bdc32f319b79.png)
Step 2.
![Screenshot 2023-03-02 at 09 53 21](https://user-images.githubusercontent.com/26574116/222319064-491e3416-c657-4353-9783-d1f3830244ca.png)
Step 3.
![Screenshot 2023-03-02 at 09 53 10](https://user-images.githubusercontent.com/26574116/222319051-85273f76-3be7-46c5-959b-bdc32f319b79.png)

**Comment by ShaMan123** [comment]
You have 2 options depending on UX:
- you can access the eraser and delete the last path
- you can use the EraserBrush#inverted option to allow the user to undo erasing
